### PR TITLE
feat: Manpages finder (output of apropos)

### PR DESCRIPTION
**Pull request by sunjon**
man

### CODE REVIEW

**Comment by sunjon** [comment]
manpages

File: lua/telescope/builtin.lua [hunk 1]
```diff
@@ -695,4 +695,43 @@ builtin.current_buffer_fuzzy_find = function(opts)
   }):find()
 end
 
+builtin.man_pages = function(opts)
+  opts = opts or {}
+
+  -- TODO: make man sections configurable?
+  local cmd = "apropos --sections=1 ''"
```
Comment (kamalmarhubi) [inline lua/telescope/builtin.lua:702]: drive-by: I'd definitely like to configure this. For my uses, a better default would be to search all sections.

**Review by kamalmarhubi — COMMENTED** [review]
this is pretty cool :-)

File: lua/telescope/builtin.lua (hunk 1 above)
Comment (tjdevries) [reply lua/telescope/builtin.lua:702]: Can be `opts.man_cmd` or similar.

File: lua/telescope/builtin.lua [hunk 2]
```diff
@@ -695,4 +695,43 @@ builtin.current_buffer_fuzzy_find = function(opts)
...
+    table.insert(lines, s)
+  end
+
+  pickers.new(opts, {
+    prompt    = 'Man',
+    finder    = finders.new_table {
+      results = lines,
+      entry_maker = make_entry.gen_from_apropos(opts),
+    },
+    -- previewer = previewers.qflist.new(opts),
```
Comment (tjdevries) [inline lua/telescope/builtin.lua:718]: Do you think we can use the Man command to be the previewed?

Comment (sunjon) [reply lua/telescope/builtin.lua:718]: yeah, I think so. I can't remember why I stopped. I think it was because the rendering of the Man page itself in nvim is still slow imo.
The whole sequence of generating an uncached list from apropos(0.5s), the slower-than-other-filetypes rendering of Man already seems slow in comparison to other finders(0.5s).

I will implement it but I think something needs to be improved to make the experience tolerable.

Comment (tjdevries) [reply lua/telescope/builtin.lua:718]: I thought it was just the first loading of a man page that takes a long time (I don't know for sure though)

**Comment by tami5** [comment]
This is extremely useful  and smart, thanks @sunjon . Hope it gets merged soon.
